### PR TITLE
Introduce qmk_id dynamic console report detection

### PR DESCRIPTION
**Pull request by t-8ch**
This utility detects the teensy debug console report descriptor.
Therefore the need to either hardcode all QMK keyboards or grant broad permissions on all hidraw devices is alleviated.

Continuation of https://github.com/qmk/qmk_firmware/pull/12980

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+PREFIX?=/usr/local
+DESTDIR?=
+
+CC = gcc
```
Comment (tzarc) [inline Makefile:4]: This allows use of `scan-build` and other static analysis tooling.
```suggestion
CC ?= gcc
```

Consequently...
```sh
scan-build $(scan-build --help-checkers | xargs -L1 echo '--enable-checker' | xargs echo) make -B
```
...comes back clean!

File: qmk_id_test.c [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#define main original_main
```
Comment (tzarc) [inline qmk_id_test.c:1]: Please copy across the license header from `qmk_id.c`.

File: qmk_id.c [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+#        define SECCOMP_ARCH_NATIVE AUDIT_ARCH_RISCV32
+#    elif __riscv_xlen == 64
+#        define SECCOMP_ARCH_NATIVE AUDIT_ARCH_RISCV64
+#    endif
+#else
+#    error Unknown target architecture
+#endif
+
+static const char *const qmk_id      = "ID_QMK=1\n";
+static const size_t      qmk_id_size = 9;
```
Comment (tzarc) [inline qmk_id.c:62]:
```suggestion
static const char qmk_id[]      = "ID_QMK=1\n";
static const size_t qmk_id_size = sizeof(qmk_id) - 1;
```

**Comment by skullydazed** [comment]
Thanks!

**Review by tzarc — COMMENTED** [review]
Guess these can come later.

**Comment by t-8ch** [comment]
> Guess these can come later.

Will send a PR tomorrow.

**Comment by tzarc** [comment]
Will sort it out on our end,  all good.

**Comment by t-8ch** [comment]
Ok, thanks!

**Pull request merged**